### DISCUSSION TITLE
docx heading property

### DISCUSSION THREAD

**Discussion by rwkmbqc** [dolanmiu/docx#2432] [Q&A] [answered]
Hi
I tried the following code below but it appears using the heading property will not work, i.e. the click button will not download the file. Other properties such as bullets work. Is there a correct way to write the heading? Hope to hear from the experts.

<!DOCTYPE html>
<html>
<head>
    <script src="https://unpkg.com/docx@5.0.2/build/index.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/FileSaver.js/2.0.5/FileSaver.js"></script>
</head>
<body>
    <h1>DOCX Generation</h1>
    <button type="button" onclick="generate()">Click to generate document</button>
    <script>
function generate() {
            const doc = new docx.Document();
            
            doc.addSection({
                properties: {},
                children: [  new docx.Paragraph({ text: "Hello YouTube", heading: HeadingLevel.TITLE }), ] }),
                docx.Packer.toBlob(doc).then(blob => {
                console.log(blob);
                saveAs(blob, "example.docx");
                console.log("Document created successfully");
            });
        }
    </script>
 </body>
</html>

**Comment by dolanmiu** [comment, marked as answer]
I do not understand the question

Refer to this for header and footers: https://github.com/dolanmiu/docx/blob/master/demo/8-header-footer.ts
